Generator/ProductGenerator.php

```diff
@@ -168,24 +189,24 @@ public function generate(array $config, $outputDir, ProgressHelper $progress, ar
             $product[$this->identifierCode] = self::IDENTIFIER_PREFIX . $i;
             $family = $this->getRandomFamily($this->faker);
             $product['family'] = $family->getCode();
+            $product['groups'] = '';
 
-            if ($nbAttrBase > 0) {
-                if ($nbAttrDeviation > 0) {
-                    $nbAttr = $this->faker->numberBetween(
-                        $nbAttrBase - round($nbAttrDeviation/2),
-                        $nbAttrBase + round($nbAttrDeviation/2)
-                    );
-                } else {
-                    $nbAttr = $nbAttrBase;
-                }
-            }
-            $familyAttrCount = count($this->getAttributesFromFamily($family));
+            /** @var VariantGroupDataProvider $variantGroupDataProvider */
+            $variantGroupDataProvider = $this->getNextVariantGroupProvider();
+            $variantGroupAttributes = [];
 
-            if (!isset($nbAttr) || $nbAttr > $familyAttrCount) {
-                $nbAttr = $familyAttrCount;
+            if (null !== $variantGroupDataProvider) {
+                $variantGroupAttributes = $variantGroupDataProvider->getAttributes();
+                $product['groups'] = $variantGroupDataProvider->getCode();
```

Comment:
we changed a bit the import/export format in 1.3, to avoid BC Break, $product['groups'] can contain both variant group and group i guess, but i would use 'variant_group' field :)